src/Controller.ino

```diff
@@ -245,10 +245,14 @@ bool MQTTCheck(int controller_idx)
     {
       if (MQTTclient_should_reconnect) {
         addLog(LOG_LEVEL_ERROR, F("MQTT : Intentional reconnect"));
+        // Homie 3 & 4dev Controller Plugin
+        // Connect first and call all installed controller to publish autodiscover data
+        // hope this is the right place ;)
+        if (MQTTConnect(controller_idx)) CPluginCall(CPLUGIN_CONNECTED, 0);
       } else {
         connectionFailures += 2;
       }
-      return MQTTConnect(controller_idx);
+      return (!MQTTclient.connected() || MQTTConnect(controller_idx)); // either connected above or now!?
```

Comment:
Not sure about performing a connect here.
Either way your change does different things here when performing a connect. 
In the first call to connect it does call the `CPLUGIN_CONNECTED` function and in the other call it doesn't.

Other MQTT controllers are reconnected just fine, so this change may change current behavior for other controllers too.

Reply:
I was not sure how `MQTTConnect(controller_idx);` works and when `MQTTclient_should_reconnect` is true. I thought that when `MQTTclient_should_reconnect `is true the function `MQTTConnect(controller_idx);` will be called twice.

Perhaps you know a better place for` CPluginCall(CPLUGIN_GOT_CONNECTED, 0);`
It should be triggered then a mqtt connection is established and stable (to send many) messages.

Reply:
I have to look.
I think just setting `MQTTclient_should_reconnect` to true should initiate a new (re)connect.

Reply:
```
      if (MQTTclient_should_reconnect) {
        addLog(LOG_LEVEL_ERROR, F("MQTT : Intentional reconnect"));
        // Connect first and call all installed controller to publish auto-discover data
        if (MQTTConnect(controller_idx)) CPluginCall(CPLUGIN_GOT_CONNECTED, 0);
      } else {
        connectionFailures += 2;
      }
      return MQTTConnect(controller_idx);
```

As I understood it on` MQTTclient_should_reconnect` it tries the first time to connect and on success it calls `CPluginCall(CPLUGIN_GOT_CONNECTED, 0);`
But on failure it will try again` return MQTTConnect(controller_idx);` but this time it will NOT call: `CPluginCall(CPLUGIN_GOT_CONNECTED, 0);` on success.

For me it`s fine. In the rare occasion this happens a manual reset helps and most of the time sending auto-discover is not necessary because the messages are retained in the (some:) broker. Only for new devices and modifications.